share/shwap/p2p/bitswap/blockstore_metrics.go

```diff
@@ -0,0 +1,212 @@
+package bitswap
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	bstore "github.com/ipfs/boxo/blockstore"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	ipld "github.com/ipfs/go-ipld-format"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const (
+	notFoundKey = "not_found"
+	failedKey   = "failed"
+)
+
+var (
+	meter                   = otel.Meter("bitswap_blockstore")
+	_     bstore.Blockstore = (*BlockstoreWithMetrics)(nil)
+)
+
+// BlockstoreWithMetrics is a blockstore that collects metrics on blockstore operations.
+type BlockstoreWithMetrics struct {
+	bs      bstore.Blockstore
+	metrics *metrics
+}
+
+type metrics struct {
+	delete      metric.Int64Counter
```

Comment:
Delete is noop in our blockstore as well and never called, dont we need it either

Reply:
Used in light node pruning.